src/Hosting/Hosting/test/Internal/HostTests.cs

```diff
@@ -266,6 +266,93 @@ public async Task StartCanBeCancelled()
             Assert.Equal(1, service.DisposeCount);
         }
 
+        [Fact]
+        public async Task CancellableStart_CancelledByApplicationLifetime()
+        {
+            var hostedService = new AsyncFakeHostedService();
+            var builder = CreateBuilder().ConfigureServices(services =>
+            {
+                services.AddSingleton<IHostedService>(hostedService);
+            });
+
+            using (var host = builder.Build())
+            {
+                var applicationLifetime = host.Services.GetService<IHostApplicationLifetime>();
+                var startTask = Task.Run(() => host.StartAsync());
```

Comment:
Task.Run isn't needed here, AsyncFakeHostedService.StartAsync doesn't block sync.

Reply:
Oh right, that’s a left-over from an early version where I was using that sync `ManualResetEvent` that the other tests use.